vulnerabilities/importers/nginx.py

```diff
@@ -178,19 +173,78 @@ def extract_vuln_pkgs(self, vuln_info):
                 )
             )
 
-        valid_versions = find_valid_versions(
-            self.version_api.get("nginx/nginx").valid_versions, version_ranges
-        )
         qualifiers = {}
         if windows_only:
             qualifiers["os"] = "windows"
 
+        purl = PackageURL(type="generic", name="nginx", qualifiers=qualifiers)
         return [
-            PackageURL(type="generic", name="nginx", version=version, qualifiers=qualifiers)
-            for version in valid_versions
+            AffectedPackage(package=purl, version_specifier=version_range)
+            for version_range in version_ranges
         ]
 
 
+class NginxTimeTravel(Improver):
+    def inferences(self):
+        self.set_api()
+        advisories = Advisory.objects.filter(
+            source="vulnerabilities.importers.nginx.NginxDataSource"
+        )
+        inferences = []
+        for advisory in advisories:
+            advisory_data = AdvisoryData.from_json(advisory.data)
+
+            affected_package_ranges = [
+                pkg.version_specifier for pkg in advisory_data.affected_packages
+            ]
+            affected_package_versions = find_valid_versions(
+                self.version_api.get("nginx/nginx").valid_versions,
+                affected_package_ranges,
+            )
+            affected_packages = []
+            for pkg in advisory_data.affected_packages:
+                for 
+                affected_packages.extend([])
+            affected_packages = [advisory_data.affected_packages]
+
+            fixed_package_ranges = [
+                pkg.version_specifier for pkg in advisory_data.affected_packages
+            ]
+            fixed_packages = find_valid_versions(
+                self.version_api.get("nginx/nginx").valid_versions, fixed_package_ranges
+            )
+
+            pkgs = nearest_patched_package(affected_package_versions, fixed_package_ranges)
+            for pkg in pkgs:
+                print(pkg)
```

Comment:
nit... you may want to guard these with a `if TRACE:` flag and keep them... otherwise drop them

Reply:
thanks!